tests/features/bootstrap/FeatureContext.php

```diff
@@ -405,22 +406,67 @@ public function assertOptionNotSelected($text) {
    *   CSS selector of the select field.
    *
    * @throws \Exception
+   *   Thrown when the select is not found in the page or the selected option is
+   *   not the expected one.
    *
    * @Then the option with text :option from select :select is selected
    */
   public function assertFieldOptionSelected($option, $select) {
-    $element = $this->findSelect($select);
+    $this->assertFieldOptionSelectedInRegion($option, $select);
+  }
+
+  /**
+   * Find the selected option of the select in a card and check the text.
+   *
+   * @param string $option
+   *   Text value of the option to find.
+   * @param string $select
+   *   CSS selector of the select field.
+   * @param string $heading
+   *   The card heading.
+   *
+   * @throws \Exception
+   *   Thrown when the select is not found in the page or the selected option is
+   *   not the expected one.
+   *
+   * @Then the option with text :option from select :select is selected in the :heading card
+   */
+  public function assertFieldOptionSelectedInCard($option, $select, $heading) {
+    $region = $this->getCollectionSubscriptionCardByHeading($heading);
+    $this->assertFieldOptionSelectedInRegion($option, $select, $region);
+  }
+
+  /**
+   * Checks whether an option is selected in a specific select in a region.
+   *
+   * @param string $option
+   *   Text value of the option to find.
+   * @param string $select
+   *   CSS selector of the select field.
+   * @param \Behat\Mink\Element\TraversableElement $region
+   *   (optional) The region to search in. Defaults to the whole page.
+   *
+   * @throws \Exception
+   *   Thrown when the select is not found in the page or the selected option is
+   *   not the expected one.
+   */
+  protected function assertFieldOptionSelectedInRegion($option, $select, TraversableElement $region = NULL) {
```

Comment:
Missed method return strict typing and strict typing for `$option` and `$select` arguments.

Starting with the word "assert" would confuse some because we usually use such namings for step definitions, but this is more a helper (code reusing). Probably `isFieldOptionSelectedInRegion`? However, it's still correct... I have no strong feeling.

Reply:
I think in this case `assertFieldOptionSelectedInRegion()` is still the better choice. It is true that we usually use this for step definitions, but it is in fact consistent with other methods that throw an exception if the expected condition is not met (for example the asserts in PHPUnit or Drupal core test traits).

The `is...` naming usually indicates that a boolean is being returned which will indicate if the condition is met. This is not the case here.

Anyway this is a small nitpick :)